File: Q-0724 - Find Pivot Index/0724.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        
        sumLeft=0
        sumRight=0
        
        for i in range(len(nums)):
            sumRight+=nums[i]
        
        for i in range(len(nums)):
            sumRight-=nums[i]
            if(sumLeft==sumRight) :                
                return i
            sumLeft+=nums[i]
                
        return -1

File: Q-0724 - Find Pivot Index/0724.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        
        int sumLeft=0, sumRight=0;
        
        for(int i=0; i<nums.size(); i++)
            sumRight+=nums[i];
        
        
        for(int i=0; i<nums.size(); i++){            
            sumRight-=nums[i];
            if(sumLeft==sumRight)                
                return i;
            sumLeft+=nums[i];
        }
                
        return -1;
    }
};